### ISSUE TITLE
Feature Request - Support Proxy Outbound

### ISSUE THREAD

**Issue by ansred** [alexjustesen/speedtest-tracker#556]
**Is your feature request related to a problem? Please describe.**
Would be nice to have proxy support, something similar to uptime kuma 

**Describe the solution you'd like**
Example screenshot:

![Screenshot 2023-04-23 at 4 09 24 PM](https://user-images.githubusercontent.com/5244872/233863221-4edfc01d-714a-445f-85e0-e9230e5b52e6.png)

**Additional context**
The reason we need this is to support monitoring a dedicated ISP on the same network, for example we run first docker instance and point to Explicit Proxy for ISP#1 and another one instance pointed to ISP#2

Thanks

**Comment by agneevX** [comment]
Proxies are not appropriate for speed tests. 

Instead, multiple macvlan interfaces can be attached to a single speedtest container and functionality could be added to speedtest that enables the customization and tests per interface.

**Comment by alexjustesen** [comment]
Agreed, I wouldn't do this with a proxy and instead do it at the networking layer.
